pkg/planner/core/operator/logicalop/logical_datasource.go

```diff
@@ -52,6 +53,8 @@ type DataSource struct {
 	Columns       []*model.ColumnInfo
 	DBName        ast.CIStr
 
+	ColIdxsByName map[string]int
```

Comment:
You can use a map `map[int]int` to store the mapping: id -> offset.
And then use the `tableInfo.Columns[IndexColumn.Offset].ID` to get the needed information.

I think the `map[string]int` will be hard to extend for other usage.

Reply:
As an example, I don't see that we update it during column pruning.